enhancements/installer/gcp-ipi-shared-vpc.md

```diff
@@ -0,0 +1,262 @@
+
+The installation process can then proceed as any other IPI workflow by running
+the create manifests, create ignition, and/or create cluster sub-commands.
+
+### API Extensions
+
+#### api.dns
+
+The current DNSZone struct does not appear to be able to store which project
+the zone is in. The ID variable can be used to store fully qualified path
+names, which will include the project. However, this may require modification
+of all consumers of this variable to be able to parse this information.
+Alternatively, a second variable can be added to store the projectID and the
+consumers can use that information when linking to the DNS Zone.
+
+#### api.ingress
+
+The current ingress structures do not appear to be able to store an address to
+be used when creating the ingress load balancer(s). New field(s) may be needed
+to enable this functionality. They can then be passed to the gcp lagacy
+provider for creating the ingress load balancers.
+
+#### installconfig.platform.gcp.networkProjectID
+
+A new parameter will be added to the install-config to specify the project id
+of the network and subnets. This parameter will then be used where necessary to
+validate existing network project resources as well as provision additional
+resources using these network resources. The default will be an empty string
+and will indicate these resources (should) exist in the default cluster
+project.
+
+#### installconfig.platform.gcp.privateManagedZone
+
+A new parameter will be added to the install-config to specify the private DNS
+managed zone. This parameter will be used where necessary to validate existing
+DNS resources as well as provision additional private DNS records. The default
+will be an empty string, which will indicate these resources are to be created
+and managed in the project identified by the networkProjectID. When this
+parameter is set to “none”, the cluster will not be configured to manage
+private DNS resources.
+
+#### installconfig.platform.gcp.publicManagedZone
+
+A new parameter will be added to the install-config to specify the public DNS
+managed zone. This parameter will be used where necessary to validate existing
+DNS resources as well as provision additional public DNS records. The default
+will be an empty string, which will indicate these resources are to be created
+and managed in the project identified by the networkProjectID (when publish is
+configured to “External”). When this parameter is set to “none”, the cluster
+will not be configured to manage public DNS resources.
+
+#### installconfig.platform.gcp.apiAddress
+
+A new parameter will be added to the install-config to specify a pre-existing
+compute address resource to be used when creating and managing the API load
+balancer(s).
+
+#### installconfig.platform.gcp.ingressAddress
+
+A new parameter will be added to the install-config to specify a pre-existing
+compute address resource to be used when creating and managing the default
+ingress load balancer(s).
```

Comment:
I think these are optional too, like the previous fields.

Reply:
Correct. Updated the text to indicate this.